llm/generate/gen_common.sh

```diff
@@ -83,8 +83,8 @@ build() {
 compress_libs() {
     echo "Compressing payloads to reduce overall binary size..."
     pids=""
-    for lib in ${BUILD_DIR}/lib/*.${LIB_EXT}* ; do
```

Comment:
@dhiltgen let me know if removing that wildcard is okay, it was creating `.gz.gz` files and so forth

Reply:
Ah, it probably isn't since some end in `.so.1`, will revisit

Reply:
The trailing `*` is needed to pick up the dependency libraries which don't all end in `.so` but have versions.

To avoid the compression inception, I'd suggest adding a `rm -rf ${BUILD_DIR}/lib/` just before
https://github.com/jmorganca/ollama/blob/main/llm/generate/gen_common.sh#L72